### PR TITLE
Override member update limit

### PR DESCRIPTION
**Pull request by kaelantdt**
Of note: the readResource file in the new test should eventually be converted to manipulation of a constant file, the file there right now is a temp file I copied.

### CODE REVIEW

File: src/main/java/org/helpberkeley/memberdata/WorkRequestHandler.java [hunk 1]
```diff
@@ -323,14 +333,17 @@ static Reply parse(Reply lastReply, final List<String> lines) {
                     version = line.replaceAll("Version:", "").trim();
                 } else if (line.toLowerCase().startsWith(DISABLE_DATE_AUDIT)) {
                     disableDateAudit = true;
-                } else if (line.contains(Constants.UPLOAD_URI_PREFIX)) {
+                } else if (line.toLowerCase().startsWith(DISABLE_MEMBER_LIMIT_AUDIT)) {
+                    disableMemberLimitAudit = true;
+                }
+                else if (line.contains(Constants.UPLOAD_URI_PREFIX)) {
```
Comment (camdetaz) [inline src/main/java/org/helpberkeley/memberdata/WorkRequestHandler.java:339]: else if should be on the line above, next to the bracket

File: src/main/java/org/helpberkeley/memberdata/WorkRequestHandler.java [hunk 2]
```diff
@@ -250,6 +253,12 @@ static class WorkRequest extends Reply {
             this.destinationTopic = reply.requestTopic;
             this.disableDateAudit = false;
             this.version = null;
+            this.disableMemberLimitAudit = false;
+
+        }
+
+        public boolean disableMemberLimitAudit(){
+            return this.disableMemberLimitAudit;
```
Comment (camdetaz) [inline src/main/java/org/helpberkeley/memberdata/WorkRequestHandler.java:261]: don't need this accessor, the boolean is not private, so you can check it with request.disableMemberLimitAudit - no parens

File: src/main/java/org/helpberkeley/memberdata/WorkflowExporter.java [hunk 3]
```diff
@@ -40,7 +41,9 @@ public class WorkflowExporter extends Exporter {
     private final WorkflowParser parser;
     private final List<WorkflowBean> updatedBeans = new ArrayList<>();
     private final Set<String> updatedUsers = new HashSet<>();
-    private int member_limit = Constants.AVG_RUN_SIZE*10;
+    public static int getDefaultMemberLimit() { return Constants.AVG_RUN_SIZE*10; }
+    public static int member_limit = getDefaultMemberLimit();
+    public static int memberLimit() { return member_limit; } //inelegant solution?
```
Comment (camdetaz) [inline src/main/java/org/helpberkeley/memberdata/WorkflowExporter.java:46]: either make member_limit private or remove the memberLimit() accessor. don't need both. if it's a public variable you can just call WorkflowExporter.member_limit without an accessor

**Review by Amosnomor — COMMENTED** [review]
This branch doesn't build:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project member-data: Compilation failure
[ERROR] /pool1/ds1-guava/stone/IdeaProjects/member-data/src/main/java/org/helpberkeley/memberdata/Main.java:[724,24] cannot find symbol
[ERROR]   symbol:   method disableMemberLimitAudit()
[ERROR]   location: variable request of type org.helpberkeley.memberdata.WorkRequestHandler.WorkRequest
```

File: src/main/java/org/helpberkeley/memberdata/WorkflowExporter.java [hunk 4]
```diff
@@ -25,6 +25,7 @@
 import java.text.MessageFormat;
 import java.util.*;
 
+
```
Comment (Amosnomor) [inline src/main/java/org/helpberkeley/memberdata/WorkflowExporter.java:28]: Remove.

File: src/main/java/org/helpberkeley/memberdata/WorkflowExporter.java [hunk 5]
```diff
@@ -40,7 +41,8 @@ public class WorkflowExporter extends Exporter {
     private final WorkflowParser parser;
     private final List<WorkflowBean> updatedBeans = new ArrayList<>();
     private final Set<String> updatedUsers = new HashSet<>();
-    private int member_limit = Constants.AVG_RUN_SIZE*10;
+    public static int getDefaultMemberLimit() { return Constants.AVG_RUN_SIZE*10; }
+    public static int member_limit = getDefaultMemberLimit();
```
Comment (Amosnomor) [inline src/main/java/org/helpberkeley/memberdata/WorkflowExporter.java:45]: Minor nit.   One of the style aspects of the code in this repository is that class CTOR definition(s) come first, before other method definitions  (though I wouldn't be surprised to find a few thoughtless exceptions in the project).  Maintaining a consistent style within a project makes it easier to read, learn, and maintain it.  In software organizations/projects there are generally coding standards that must be met.

Here, I would either move getDefaultMemberLimit() next to changeMemberLimit(), or yet simpler, just define it as a static final constant:
```public static final int DEFAULT_MEMBER_LIMIT = Constants.AVG_RUN_SIZE * 10 ;```

Another style point is that [camelCase](https://en.wikipedia.org/wiki/Camel_case) naming is used throughout this project. Change this to to memberLimit;

Change memberLimit to be private, and add an accessor for it to be used by the test case.

File: src/main/java/org/helpberkeley/memberdata/WorkflowExporter.java [hunk 6]
```diff
@@ -179,7 +181,8 @@ private String updatedWorkflowToString() {
         return updatedData.toString();
     }
 
-    public void changeMemberLimit(int limit) {
+    public static void changeMemberLimit(int limit) {
```
Comment (Amosnomor) [inline src/main/java/org/helpberkeley/memberdata/WorkflowExporter.java:184]: In Java accessors are typically named getXXX and setXXX.

**Review by Amosnomor — COMMENTED** [review]
Hello Kaelan.  Welcome to the project!

File: src/test/java/org/helpberkeley/memberdata/MainTest.java [hunk 7]
```diff
@@ -866,6 +866,28 @@ public void updateMemberDataNoUpdatesTest() throws IOException, CsvException {
...
+        HttpClientSimulator.setQueryResponseData(
+                Constants.QUERY_GET_REQUESTS_LAST_REPLIES, request);
+        String usersFile = findFile(Constants.MEMBERDATA_RAW_FILE, "csv");
+        String[] args = { Options.COMMAND_WORK_REQUESTS, usersFile };
+        try {
+            WorkflowExporter.setMemberLimit(2);
+            Main.main(args);
+            assertThat(WorkflowExporter.getMemberLimit() == 15); //size of user list in users.json
+        } finally {
+            WorkflowExporter.setMemberLimit(WorkflowExporter.DEFAULT_MEMBER_LIMIT);
```
Comment (Amosnomor) [inline src/test/java/org/helpberkeley/memberdata/MainTest.java:884]: Now that we've made this static we need this same finally treatment in every test that uses setMemberLimit().

**Review by Amosnomor — COMMENTED** [review]
Looks good.  Just one other thing to fix and then I can merge this

**Comment by kaelantdt** [comment]
In the last tests, I noticed a warning that came up twenty times or so. It reads (with some variation): 
12:01:17.035 [main] WARN  o.helpberkeley.memberdata.ApiClient - send https://go.helpberkeley.org/admin/plugins/explorer/queries/11/run POST failed: We are having technical difficulties
12:01:17.037 [main] WARN  o.helpberkeley.memberdata.ApiClient - Failure talking to Discourse, waiting 10 seconds and retrying.

If I recall correctly, this has come up when running tests in the past, but I wanted to know what exactly is going on here. When I get warnings on tests, I figure it is good to get a baseline on what is concerning, and what is not.

**Comment by Amosnomor** [comment]
> In the last tests, I noticed a warning that came up twenty times or so. It reads (with some variation): 12:01:17.035 [main] WARN o.helpberkeley.memberdata.ApiClient - send https://go.helpberkeley.org/admin/plugins/explorer/queries/11/run POST failed: We are having technical difficulties 12:01:17.037 [main] WARN o.helpberkeley.memberdata.ApiClient - Failure talking to Discourse, waiting 10 seconds and retrying.
> 
> If I recall correctly, this has come up when running tests in the past, but I wanted to know what exactly is going on here. When I get warnings on tests, I figure it is good to get a baseline on what is concerning, and what is not.

These messages are expected when running all of the Maven based tests. There are test cases which simulate Discourse failures, which are typically due to request rate throttling imposed by Discourse.  See ApiClient.send() for the retry logic we have in place for dealing with this.  These messages are showing up in test output because we log these Discourse failures as warnings in the production code. We want to know about these when they occur in the production system.  But we could also do something to not have them pollute the test output (perhaps something programatic the tests could do to change the log level).  Ideally tests should be silent except when failing.

**Review by Amosnomor — APPROVED** [review]
Thanks.  I will merge now.

**Pull request merged**